Avoid implicitly declaring exit.  Future compilers will not support
implicit function declarations, so this cmake probe will always fail.

Submitted upstream: <https://sourceforge.net/p/plplot/patches/37/>

diff --git a/cmake/modules/TestForHighBitCharacters.c b/cmake/modules/TestForHighBitCharacters.c
index 338cc5ef72d90719..1dbcc1c54288ea6d 100644
--- a/cmake/modules/TestForHighBitCharacters.c
+++ b/cmake/modules/TestForHighBitCharacters.c
@@ -17,6 +17,6 @@ int
    int i;
    for (i = 0; i < 256; i++)
      if (XOR (islower (i), ISLOWER (i)) || toupper (i) != TOUPPER (i))
-       exit(1);
-   exit (0);
+       return 1;
+   return 0;
 }
